File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/dao/InventoryDao.java
package com.firefly.example.reactive.coffee.store.dao;

import com.firefly.example.reactive.coffee.store.model.Inventory;
import com.firefly.example.reactive.coffee.store.vo.InventoryOperator;
import com.firefly.example.reactive.coffee.store.vo.InventoryUpdate;
import reactor.core.publisher.Mono;

import java.util.List;

/**
 * @author Pengtao Qiu
 */
public interface InventoryDao {

    Mono<int[]> updateBatch(List<InventoryUpdate> list, InventoryOperator operator);

    Mono<List<Inventory>> listByProductId(List<Long> productIdList);

}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/dao/InventoryDAO.kt
package com.firefly.example.kotlin.coffee.store.dao

import com.firefly.example.kotlin.coffee.store.model.Inventory
import com.firefly.example.kotlin.coffee.store.vo.InventoryOperator
import com.firefly.example.kotlin.coffee.store.vo.InventoryUpdate

/**
 * @author Pengtao Qiu
 */
interface InventoryDAO {

    suspend fun updateBatch(list: List<InventoryUpdate>, operator: InventoryOperator): IntArray

    suspend fun listByProductId(productIdList: List<Long>): List<Inventory>

}